File: CHUONG_5/5.15_ViewPager_Exercise/app/src/main/java/com/example/viewpager/data/model/Zodiac.java
package com.example.viewpager.data.model;

import java.util.Objects;

public class Zodiac {
    private String name;
    private Integer drawableId;

    public Zodiac() {}
    public Zodiac(String name, Integer drawableId){
        this.name = name;
        this.drawableId = drawableId;
    }

    public String getName() {
        return name;
    }

    public Integer getDrawableId() {
        return drawableId;
    }

    public void setDrawableId(Integer drawableId) {
        this.drawableId = drawableId;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public boolean equals(Object o) {
        if (o == null || getClass() != o.getClass()) return false;
        Zodiac zodiac = (Zodiac) o;
        return Objects.equals(drawableId, zodiac.drawableId);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(drawableId);
    }
}

File: CHUONG_5/5.15_Zodiac_kotlin/app/src/main/java/com/example/zodiac/data/model/Zodiac.kt
package com.example.zodiac.data.model

data class Zodiac(
    val name: String,
    val drawableId: Int?
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Zodiac
        return drawableId == other.drawableId
    }

    override fun hashCode(): Int {
        return drawableId ?: 0
    }
}